syncopy/nwanalysis/wilson_sf.py

```diff
@@ -59,17 +63,25 @@ def wilson_sf(CSD, nIter=100, rtol=1e-9):
     psi = np.tile(psi0, (nFreq, 1, 1))
     assert psi.shape == CSD.shape
 
+    g = np.zeros(CSD.shape, dtype=np.complex64)
     converged = False
     for _ in range(nIter):
 
-        psi_inv = np.linalg.inv(psi)
-        # the bracket of equation 3.1
-        g = psi_inv @ CSD @ psi_inv.conj().transpose(0, 2, 1)
+        if direct_inversion:
+            psi_inv = np.linalg.inv(psi)
+            # the bracket of equation 3.1
+            g = psi_inv @ CSD @ psi_inv.conj().transpose(0, 2, 1)
+        else:
+            for i in range(nFreq):
```

Comment:
I think we don't need the for loop: `numpy.linalg.lstsq` can process right hand sides `b` with shape `(M,M,K)`, right? One transpose of `CSD` should do it and we don't have to iterate over frequencies.

Reply:
No they actually can not, shapes are: 
```
a : (M, N) array_like
    "Coefficient" matrix.
b : {(M,), (M, K)} array_like
```

Reply:
I see, my bad - the coefficient matrix hast to be 2d, the rhs can be 3d. Never mind then, thanks for the clarification.